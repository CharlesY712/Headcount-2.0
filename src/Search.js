import React, { Component } from 'react';
// import PropTypes from 'prop-types';

class Search extends Component {
  constructor(props) {
    super(props);
    this.state = {
      location: ''
    };
  }
  
  handleChange = (event) => {
    this.setState({location: event.target.value.toUpperCase()});
    this.props.findMatch(event.target.value);
  }

  render() {
    return (
      <div>
        <input 
          type='text'
          placeholder='Search location'
          value={this.state.location}
          onChange={this.handleChange}
        />
        <button 
          type='submit'
        >Clear</button>
      </div>
    );
  }
}

export default Search;
