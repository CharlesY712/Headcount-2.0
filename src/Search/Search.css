.search-bar {
  text-align: center
}

.search-input {
  width: 60%;
  height: 40px;
  font-size: 30px;
  margin-bottom: 20px;
}

input::placeholder {
  font-size: 30px;
}

.clear-button {
  height: 45px;
  width: 100px;
  vertical-align: top;
  border-radius: 30px 30px 30px 30px;
  margin-left: 5px;
  font-size: 2em;
  color: #6A6A6A;
}

.clear-button:hover {
  background-color: #45A2CA;
  color: black;
}

.description {
  margin-top: 0;
  margin-bottom: 10px;
  font-size: 20px;
}