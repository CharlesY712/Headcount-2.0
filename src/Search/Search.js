import React, { Component } from 'react';
import './Search.css';
// import PropTypes from 'prop-types';

class Search extends Component {
  constructor(props) {
    super(props);
    this.state = {
      location: ''
    };
  }
  
  handleChange = (event) => {
    this.setState({location: event.target.value});
    this.props.findMatch(event.target.value);
  }

  handleSubmit = () => {
    this.setState({location: ''});
    this.props.findMatch();
  }

  render() {
    return (
      <div className= 'search-bar'>
        <input 
          className='search-input'
          type='text'
          placeholder='Search location'
          value={this.state.location}
          onChange={this.handleChange}
        />
        <button 
          className='clear-button'
          type='submit'
          onClick={this.handleSubmit}
        >Clear</button>
      </div>
    );
  }
}

export default Search;
