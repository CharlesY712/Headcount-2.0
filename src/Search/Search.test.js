import React from 'react';
import Search from './Search';
import { shallow } from 'enzyme';

describe('Search', () => {

  /* eslint-disable no-undef*/
  const mockFindMatch = jest.fn();
  /* eslint-enable no-undef*/

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Search findMatch={mockFindMatch}/>);
  });

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('updates state of location on handle change', () => {
    const mockEvent = {target: {value: 'abc'}};
    const expectedState = {location: 'abc'};

    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state()).toEqual(expectedState);
  });

  it('should invoke findMatch on handle change', () => {
    const mockEvent = { target: { value: 'abc' } };
    wrapper.instance().handleChange(mockEvent);
    expect(mockFindMatch).toHaveBeenCalled();
  });
  
  it('updates state of location on handleSubmit', () => {
    const expectedState = { location: '' };

    wrapper.setState({location: 'colorado'});
    expect(wrapper.state()).toEqual({location: 'colorado'});
    wrapper.instance().handleSubmit();
    expect(wrapper.state()).toEqual(expectedState);
  });

  it('should invoke findMatch on handleSubmit', () => {
    wrapper.instance().handleSubmit();
    expect(mockFindMatch).toHaveBeenCalled();
  });
});