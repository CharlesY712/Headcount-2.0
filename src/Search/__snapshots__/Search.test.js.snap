// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    findMatch={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="search-input"
          onChange={[Function]}
          placeholder="Search location"
          type="text"
          value=""
/>,
        <button
          className="clear-button"
          onClick={[Function]}
          type="submit"
>
          Clear
</button>,
      ],
      "className": "search-bar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "search-input",
          "onChange": [Function],
          "placeholder": "Search location",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Clear",
          "className": "clear-button",
          "onClick": [Function],
          "type": "submit",
        },
        "ref": null,
        "rendered": "Clear",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="search-input"
            onChange={[Function]}
            placeholder="Search location"
            type="text"
            value=""
/>,
          <button
            className="clear-button"
            onClick={[Function]}
            type="submit"
>
            Clear
</button>,
        ],
        "className": "search-bar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "search-input",
            "onChange": [Function],
            "placeholder": "Search location",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Clear",
            "className": "clear-button",
            "onClick": [Function],
            "type": "submit",
          },
          "ref": null,
          "rendered": "Clear",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
